src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp

```diff
@@ -117,6 +147,9 @@ namespace OpenMS
     while (MRMDecoy::AASequenceIdentity(peptide.sequence, shuffled.sequence) > identity_threshold &&
            attempts < max_attempts)
     {
+      // Block tryptic residues and N-/C-terminus from shuffling
+      IndexType idx = MRMDecoy::find_all_tryptic_and_term(peptide.sequence);
```

Comment:
For the ``shuffle`` method, we now also exclude N- and C-terminal residues from being shuffled. This has the benefit that if the terminal residues are modified, we don't have to worry about these modifications being shuffled.